sdks/python/apache_beam/testing/test_stream.py

```diff
@@ -163,45 +167,71 @@ class TestStream(PTransform):
   """Test stream that generates events on an unbounded PCollection of elements.
 
   Each event emits elements, advances the watermark or advances the processing
-  time.  After all of the specified elements are emitted, ceases to produce
+  time. After all of the specified elements are emitted, ceases to produce
   output.
   """
 
   def __init__(self, coder=coders.FastPrimitivesCoder(), events=()):
     super(TestStream, self).__init__()
     assert coder is not None
     self.coder = coder
-    self.current_watermark = timestamp.MIN_TIMESTAMP
-    self.events = list(events)
+    self.watermarks = {None: timestamp.MIN_TIMESTAMP}
+    self._events = list(events)
+    self.output_tags = set()
 
   def get_windowing(self, unused_inputs):
     return core.Windowing(window.GlobalWindows())
 
   def expand(self, pbegin):
     assert isinstance(pbegin, pvalue.PBegin)
     self.pipeline = pbegin.pipeline
-    return pvalue.PCollection(self.pipeline, is_bounded=False)
 
-  def _infer_output_coder(self, input_type=None, input_coder=None):
-    return self.coder
+    # This multiplexing the  multiple output PCollections.
+    def mux(event):
+      if event.tag:
+        yield pvalue.TaggedOutput(event.tag, event)
+      else:
+        yield event
+    mux_output = (pbegin
+                  | _TestStream(self.output_tags, events=self._events)
+                  | 'TestStream Multiplexer' >> beam.ParDo(mux).with_outputs())
```

Comment:
You don't need to pass the list of outputs to `with_outputs`?

Reply:
Nope! The DoOutputsTuple dynamically adds the outputs when you use the [ ] operator.